File: system-test/java/src/main/java/com/mycompany/myshop/testkit/dsl/port/myshop/then/steps/ThenOrder.java
package com.mycompany.myshop.testkit.dsl.port.myshop.then.steps;

import com.mycompany.myshop.testkit.driver.port.myshop.dtos.OrderStatus;
import com.mycompany.myshop.testkit.dsl.port.myshop.then.steps.base.ThenStep;

public interface ThenOrder extends ThenStep<ThenOrder> {
    ThenOrder hasSku(String expectedSku);

    ThenOrder hasQuantity(int expectedQuantity);

    ThenOrder hasUnitPrice(double expectedUnitPrice);

    ThenOrder hasBasePrice(double expectedBasePrice);

    ThenOrder hasBasePrice(String expectedBasePrice);

    ThenOrder hasSubtotalPrice(double expectedSubtotalPrice);

    ThenOrder hasSubtotalPrice(String expectedSubtotalPrice);

    ThenOrder hasTotalPrice(double expectedTotalPrice);

    ThenOrder hasTotalPrice(String expectedTotalPrice);

    ThenOrder hasStatus(OrderStatus expectedStatus);

    ThenOrder hasTotalPriceGreaterThanZero();

    ThenOrder hasOrderNumberPrefix(String expectedPrefix);

    ThenOrder hasDiscountRate(double expectedDiscountRate);

    ThenOrder hasDiscountAmount(double expectedDiscountAmount);

    ThenOrder hasDiscountAmount(String expectedDiscountAmount);

    ThenOrder hasAppliedCouponCode(String expectedCouponCode);

    ThenOrder hasAppliedCoupon(String expectedCouponCode);

    ThenOrder hasAppliedCoupon();

    ThenOrder hasTaxRate(double expectedTaxRate);

    ThenOrder hasTaxRate(String expectedTaxRate);

    ThenOrder hasTaxAmount(String expectedTaxAmount);
}

File: system-test/typescript/src/testkit/dsl/port/myShop/then/steps/then-order.ts
export interface ThenOrder extends PromiseLike<void> {
  and(): this;
  hasSku(expectedSku: string): this;
  hasQuantity(expectedQuantity: number): this;
  hasUnitPrice(expectedUnitPrice: number): this;
  hasBasePrice(basePrice: string | number): this;
  hasSubtotalPrice(subtotalPrice: string | number): this;
  hasTotalPrice(totalPrice: string | number): this;
  hasStatus(expectedStatus: string): this;
  hasTotalPriceGreaterThanZero(): this;
  hasOrderNumberPrefix(expectedPrefix: string): this;
  hasDiscountRate(expectedDiscountRate: number): this;
  hasDiscountAmount(expectedDiscountAmount: string | number): this;
  hasAppliedCoupon(expectedCouponCode?: string | null): this;
  hasTaxRate(expectedTaxRate: string | number): this;
  hasTaxAmount(expectedTaxAmount: string | number): this;
}
